### PR TITLE
OSDOCS:998 declarative configs for NE

### PR DESCRIPTION
**Pull request by skrthomas**
Please review the following:
-[Scaling an Ingress Controller](https://deploy-preview-32924--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator?utm_source=github&utm_campaign=bot_dp#nw-ingress-controller-configuration_configuring-ingress)

-[Setting a custom default certificate](https://deploy-preview-32924--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator?utm_source=github&utm_campaign=bot_dp#nw-ingress-setting-a-custom-default-certificate_configuring-ingress)

-[Configuring the route admission policy](https://deploy-preview-32924--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator?utm_source=github&utm_campaign=bot_dp#nw-route-admission-policy_configuring-ingress)

-[Enabling HTTP/2 Ingress connectivity](https://deploy-preview-32924--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator?utm_source=github&utm_campaign=bot_dp#nw-http2-haproxy_configuring-ingress)

Jira story for doc work: https://issues.redhat.com/browse/OSDOCS-2240

`oc patch` and `oc annotate` instances are followed by a TIP that suggests a YAML manifest can be applied as an alternative.

### CODE REVIEW

File: modules/nw-ingress-setting-a-custom-default-certificate.adoc [hunk 1]
```diff
@@ -86,3 +86,20 @@ The certificate secret name should match the value used to update the CR.
...
+----
+apiVersion: v1
+kind: IngressController
+metadata:
+  name:default
+  namespace:openshift-ingress-operator
+spec:
+  defaultCertificate:
+    name: custom-certs-default
+
```
Comment (jboxman) [inline modules/nw-ingress-setting-a-custom-default-certificate.adoc:103]: @skrthomas, I'd omit the trailing space here.

File: modules/nw-scaling-ingress-controller.adoc [hunk 2]
```diff
@@ -52,8 +57,19 @@ $ oc get -n openshift-ingress-operator ingresscontrollers/default -o jsonpath='{
 ----
 3
 ----
-
-[NOTE]
++
+[TIP]
 ====
-Scaling is not an immediate action, as it takes time to create the desired number of replicas.
+You can alternatively apply the following YAML to scale an Ingress Controller:
```
Comment (candita) [inline modules/nw-scaling-ingress-controller.adoc:63]: Maybe clarify this by pointing out the change is to the `replicas` value, e.g.:
You can alternatively apply the following YAML to scale an Ingress Controller to three replicas.  If you desire fewer or more than three, change the `replicas` value.

File: modules/nw-scaling-ingress-controller.adoc [hunk 3]
```diff
@@ -52,8 +57,19 @@ $ oc get -n openshift-ingress-operator ingresscontrollers/default -o jsonpath='{
...
+[source,yaml]
+----
+apiVersion: v1
+kind: IngressController
+metadata:
+  name:default
+  namespace:openshift-ingress-operator
+spec:
+  replicas: 3
+    name: custom-certs-default
```
Comment (candita) [inline modules/nw-scaling-ingress-controller.adoc:73]: Line 73 is probably not needed here.

File: modules/nw-ingress-setting-a-custom-default-certificate.adoc [hunk 4]
```diff
@@ -82,6 +82,22 @@ $ oc get --namespace openshift-ingress-operator ingresscontrollers/default \
...
+[source,yaml]
+----
+apiVersion: v1
+kind: IngressController
+metadata:
+  name:default
+  namespace:openshift-ingress-operator
+spec:
+  defaultCertificate:
+    name: custom-certs-default
```
Comment (Miciah) [inline modules/nw-ingress-setting-a-custom-default-certificate.adoc:97]: Need to fix the API group name and add spaces after `name:` and `namespace:`:
```suggestion
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name: default
  namespace: openshift-ingress-operator
spec:
  defaultCertificate:
    name: custom-certs-default
```

File: modules/nw-route-admission-policy.adoc [hunk 5]
```diff
@@ -34,3 +34,20 @@ spec:
...
+----
+apiVersion: v1
+kind: IngressController
+metadata:
+  name:default
+  namespace:openshift-ingress-operator
+spec:
+  replicas: 2
+  routeAdmission:
+    namespaceOwnership: InterNamespaceAllowed
```
Comment (Miciah) [inline modules/nw-route-admission-policy.adoc:51]: Need to fix the API group name and add spaces after `name:` and `namespace:`, and presumably setting `replicas: 2` isn't supposed to be part of the pocedure, so it should be deleted:
```suggestion
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name: default
  namespace: openshift-ingress-operator
spec:
  routeAdmission:
    namespaceOwnership: InterNamespaceAllowed
```

File: modules/nw-scaling-ingress-controller.adoc (hunk 3 above)
Comment (Miciah) [inline modules/nw-scaling-ingress-controller.adoc:73]: Need the API group and spaces, and `name: custom-certs-default` looks like leftover copypasta:
```suggestion
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name: default
  namespace: openshift-ingress-operator
spec:
  replicas: 3
```
By the way, it is also possible to use `oc scale`:
```shell
oc scale -n openshift-ingress-operator ingresscontroller/default --replicas=3
```

**Comment by ahardin-rh** [comment]
LGTM!

**Comment by skrthomas** [comment]
@lihongan this PR is ready for QA. Can you please take a look?

**Comment by lihongan** [comment]
LGTM.
Thank you @skrthomas

**Pull request merged**